core/environment/environment.go

```diff
@@ -834,3 +847,44 @@ func (env *Environment) GetVarsAsString() string {
 	}
 	return sortMapToString(varStack)
 }
+
+// return true if the auto stop transition has been scheduled, false otherwise
+// if true, the expected stop time is also returned
+func (env *Environment) scheduleAutoStopTransition() (bool, time.Time) {
+	autoStopEnabled, ok := env.UserVars.Get("auto_stop_enabled")
```

Comment:
This way you're only querying the `UserVars` (exclusively user input) of the global context, so you risk missing some of the variable sources. Instead I recommend using `env.GetKV("" /*empty path*/, "auto_stop_enabled")`.